darkstar_install_script.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+# Make sure its running with root
+if [[ $EUID != 0 ]]; then
+  printf "Root permissions required, starting with sudo\n\n"
+  chmod +x "$0"
+  sudo "$0" "$@"
+  exit $?
+fi
+# Static script Options
+APT_GET_INSTALL_ONE="nano git mariadb-client mariadb-server libmariadb-dev libmariadbclient-dev libssl-dev"
+APT_GET_INSTALL_TWO="luajit lua5.1 liblua5.1-dev libzmq3-dev autoconf pkg-config gcc build-essential"
+INSTALL_DIR="/opt/darkstar"
+# User set options (These are replaced by user input later in the script)
+DARKSTAR_USER="darkstar"
+DARKSTAR_PASSWORD="darkstar"
+DARKSTAR_IP="127.0.0.1"
+# Start of Script
+clear
+printf "This script will install Darkstar's FFXI Private Server."
+printf "\nSupported Operating Systems: Debian, Ubuntu, Raspbian\n\n"
+printf "1. Install Darkstar & Dependencies\n"
+printf "2. Update Configurations Only\n"
+printf "3. Restart Darkstar Server\n"
+printf "4. Shutdown Darkstar Server & Disable Auto Start\n"
+printf "5. Start Darkstar Server & Enable Auto Start\n"
+printf "6. Exit\n\n"
+read -p "Please enter the number of one of the above Options: " -n 1 -r RUN_CHOICE
+if [ "${RUN_CHOICE}" == "1" ]; then
+  if [ -f /opt/darkstar/dsgame ]; then
+    printf "\n\nPrevious install detected - Upgrades are not supported\nExiting script\n"
+    exit
+  fi
+  printf "\n\n1. Stable Version\n"
+  printf "2. Development Version\n\n"
+  read -p "Would you like to install the Stable or Development versions?: " -n 1 -r GIT_VERSION
+  printf "\n\nEnter a username for running the server and logging into the SQL database.\n"
+  read -r -p "Username: " DARKSTAR_USER
+  echo
+  read -r -p "'${DARKSTAR_USER}' User's Password: " DARKSTAR_PASSWORD
+  OS=$(awk '/^ID=/' /etc/os-release | sed 's/ID=//')
+  printf "\nDetected OS: %s" "${OS}"
+  # shellcheck disable=SC2086
+  apt-get update && apt-get -y install ${APT_GET_INSTALL_ONE} ${APT_GET_INSTALL_TWO}
+  adduser "${DARKSTAR_USER}" --gecos "Darkstar Server" --disabled-password
+  if [[ ${GIT_VERSION} == "1" ]]; then
+    git clone --branch stable --recursive https://github.com/DarkstarProject/darkstar/ /opt/darkstar
+  else
+    git clone --recursive https://github.com/DarkstarProject/darkstar/ /opt/darkstar
+  fi
+  cd /opt/darkstar || exit
+  sh autogen.sh
+  if [ "${OS}" == "debian" ] || [ "${OS}" == "raspbian" ]; then
+    printf "\nRunning Debian or Raspbian Install\n"
+    apt-get -y install libmariadb-dev-compat
+    ./configure CXXFLAGS=" -pthread"
+  else
+    printf "\nRunning Default Install\n"
+    ./configure
```

Comment:
Check if it completes successfully?

Reply:
Do the ./configure, make and autogen.sh return a none 0 value when they fail? I should be able to add a `|| exit` if they do.

Reply:
I was able to test it, and it does indeed exit on fail with `|| exit` and continues on success. 
Now set to exit on failure.